adapter/input/http/handler/address_test.go

```diff
@@ -0,0 +1,106 @@
+package handler
+
+import (
+	"bytes"
+	"encoding/json"
+	"fmt"
+	"github.com/petshop-system/petshop-api/application/domain"
+	"github.com/petshop-system/petshop-api/application/port/output"
+	"github.com/petshop-system/petshop-api/application/service"
+	"github.com/petshop-system/petshop-api/application/utils"
+	"github.com/stretchr/testify/assert"
+	"go.uber.org/zap"
+	"net/http"
+	"net/http/httptest"
+	"testing"
+	"time"
+)
+
+func TestAddress_Create(t *testing.T) {
+
+	tests := []struct {
+		name            string
+		addressRequest  AddressRequest
+		mockRepository  output.IAddressDomainDataBaseRepository
+		cacheRepository output.IAddressDomainCacheRepository
+		expectedCode    int
+		expectedBody    string
+	}{
+		{
+			name: "Test Successful - Create",
+			addressRequest: AddressRequest{
+				Street:       utils.MockAddressStreet,
+				Number:       utils.MockAddressNumber,
+				Complement:   utils.MockAddressComplement,
+				Neighborhood: utils.MockAddressNeighborhood,
+				ZipCode:      utils.MockAddressZipCode,
+				City:         utils.MockAddressCity,
+				State:        utils.MockAddressState,
+				Country:      utils.MockAddressCountry,
+			},
+			mockRepository: output.AddressDomainDataBaseRepositoryMock{
+				SaveMock: func(contextControl domain.ContextControl, address domain.AddressDomain) (domain.AddressDomain, error) {
+					address = utils.GetMockAddress()
+					address.ID = 0
```

Comment:
Num mundo ideal o retorno seria 0? É válido sempre manter próximo ao mundo real. 

E se o retorno for algo acima de 0 (zero)? 🤔

OBS.: necessita mudança.